astrolabe/validator.py

```diff
@@ -18,33 +18,27 @@
 from unittest import TestCase
 
 from pymongo import MongoClient
+import yaml
 
 from atlasclient import JSONObject
 from astrolabe.exceptions import WorkloadExecutorError
 from astrolabe.utils import DriverWorkloadSubprocessRunner, load_test_data
 
 
-DRIVER_WORKLOAD = JSONObject.from_dict({
-    'database': 'validation_db',
-    'collection': 'validation_coll',
-    'testData': [{'_id': 'validation_sentinel', 'count': 0}],
-    'operations': []
-})
-
-
 class ValidateWorkloadExecutor(TestCase):
     WORKLOAD_EXECUTOR = None
     CONNECTION_STRING = None
     STARTUP_TIME = None
 
     def setUp(self):
         self.client = MongoClient(self.CONNECTION_STRING, w='majority')
-        self.coll = self.client.get_database(
-            DRIVER_WORKLOAD['database']).get_collection(
-            DRIVER_WORKLOAD['collection'])
-        load_test_data(self.CONNECTION_STRING, DRIVER_WORKLOAD)
 
     def run_test(self, driver_workload):
+        self.coll = self.client.get_database(
+            [e for e in driver_workload['createEntities'] if 'database' in e][0]['database']['databaseName']).get_collection(
```

Comment:
This is really hard to read/follow. I also think this will break if there are multiple DBs created in createEntities (not that this matters since we don't seem to currently be doing this)?
Instead, I suggest spelling out the logic explicitly so that its easy to read and its limitation are self-evident:

```
dbname = None
collname = None
for e in driver_workload['createEntities']:
    if dbname and collname:
        break
    if dbname is None and 'database' in e:
        dbname = e['database']['databaseName']
    elif collanme is None and 'collection' in e:
        collname = e['collection']['collectionName']

if not (dbname and collname):
    self.fail('Invalid scenario: executor validator test cases must provide database and collection entities')

self.coll = self.client.get_database(dbname).get_collection(collname)
```

Reply:
Okay I changed to this version.